inject/service/src/main/java/io/helidon/inject/service/Injection.java

```diff
@@ -0,0 +1,265 @@
+     */
+    @Retention(RetentionPolicy.CLASS)
+    @Target({ElementType.METHOD, ElementType.CONSTRUCTOR, ElementType.FIELD})
+    @Documented
+    public @interface Inject {
+    }
+
+    /**
+     * Marks annotations that act as qualifiers.
+     * <p>
+     * A qualifier annotation restricts the eligible service instances that can be injected into an injection point to those
+     * qualified by the same qualifier.
+     */
+    @Target(ElementType.ANNOTATION_TYPE)
+    @Retention(RetentionPolicy.CLASS)
+    @Documented
+    public @interface Qualifier {
+    }
+
+    /**
+     * A qualifier that can restrict injection to specifically named instances, or that qualifies services with that name.
+     */
+    @Qualifier
+    @Retention(RetentionPolicy.CLASS)
+    @Documented
+    @Target({ElementType.CONSTRUCTOR, ElementType.METHOD, ElementType.PARAMETER, ElementType.FIELD, TYPE})
+    public @interface Named {
+        /**
+         * Type name of this annotation.
+         */
+        TypeName TYPE_NAME = TypeName.create(Named.class);
+        /**
+         * Represents a wildcard name (i.e., matches anything).
+         */
+        String WILDCARD_NAME = "*";
+        /**
+         * Default name to identify a default instance.
+         */
+        String DEFAULT_NAME = "@default";
+
+        /**
+         * The name.
+         *
+         * @return name this injection point requires, or this service provides, or a supplier provider
+         */
+        String value();
+    }
+
+    /**
+     * Annotation to force generation of a service descriptor, even if it otherwise would not be created.
+     * <p>
+     * Helidon generates service descriptors for types that have the type, field, constructor, method, or a parameter annotated
+     * with any of the supported annotations (such as {@link Injection}.
+     * This annotation can be used if we want to use a POJO with an accessible no-args constructor and no other
+     * annotations as a service, as otherwise no descriptor would be generated.
+     */
+    @Documented
+    @Retention(RetentionPolicy.CLASS)
+    @Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE})
+    public @interface Service {
+    }
+
+    /**
+     * A singleton service.
+     * The service registry will only contain a single instance of this service, and all injection points will be satisfied by
+     * the same instance.
+     * <p>
+     * A singleton instance is guaranteed to have its constructor, post-construct, and pre-destroy methods invoked once within
+     * the lifecycle of the service registry.
+     */
+    @Documented
+    @Retention(RetentionPolicy.CLASS)
+    @Service
+    @Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+    public @interface Singleton {
+        /*
+        Implementation note: we currently do not support custom scopes, so there is no Scope meta annotation.
+        If we decide to support scopes, we may want to introduce such an annotation.
+        */
+    }
```

Comment:
What's the default scope if `@Singleton` isn't specified? Is that explained elsewhere?

Reply:
Default is the same as annotated with `@Service`, I will update the docs (new instance injected per call to provider)